datalad_next/create_sibling_webdav.py

```diff
@@ -0,0 +1,157 @@
+            constraints=EnsureStr()),
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to process.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        name=Parameter(
+            args=('-s', '--name',),
+            metavar='NAME',
+            doc="Name of the sibling.",
+            constraints=EnsureStr() | EnsureNone(),
+            required=True),
+        credential=Parameter(
+            args=("--credential",),
+            doc="""Name of the credentials that should be used to access
+            the WebDAV server.""",
+        )
+    )
+
+    @staticmethod
+    @datasetmethod(name='create_sibling_webdav')
+    @eval_results
+    def __call__(
+            url: str,
+            *,
+            dataset: Optional[Union[str, Dataset]] = None,
+            name: Optional[str] = None,
+            credential: Optional[str] = None):
+        """
+
+        :param url:
+        :param dataset:
+        :param name:
+        :param credential:
+        :return: a generator yielding result records
+        """
+
+        ds = require_dataset(
+            dataset or ".",
+            check_installed=True,
+            purpose=f'create WebDAV sibling(s)')
+
+        res_kwargs = dict(
+            action=f'create_sibling_webdav',
+            logger=lgr,
+            refds=ds.path,
+        )
+
+        if not isinstance(ds.repo, AnnexRepo):
+            yield {
+                **res_kwargs,
+                "status": "error",
+                "msg": "require annex repo to create WebDAV sibling"
+            }
+            return
+
+        parser_url = urlparse(url)
+        if parser_url.query:
+            yield {
+                **res_kwargs,
+                "status": "error",
+                "msg": "URLs with query are not yet supported"
+            }
+            return
+
+        name_base = name or parser_url.hostname
+        git_name = name_base + "-wd-vcs"
+        annex_name = name_base + "-wd-tree"
+
+        datalad_annex_url = (
+            "datalad_annex::"
+            + url
+            + "?type=webdav&url={noquery}&encryption=none&exporttree=yes"
+            + "" if credential is None else f"&dlacredential={credential}"
+        )
+
+        # Add the datalad_annex:: sibling to datalad
+        from datalad.api import siblings
+
+        siblings(
+            action="add",
+            dataset=ds,
+            name=git_name,
+            url=datalad_annex_url)
+
+        # Add a git-annex webdav special remote. This might requires to set
+        # the webdav environment variables accordingly.
+
+        credential_holder = UserPassword(name=credential)()
+        os.environ["WEBDAV_USERNAME"] = credential_holder["user"]
+        os.environ["WEBDAV_PASSWORD"] = credential_holder["password"]
```

Comment:
we better use `with patch.dict('os.environ', ...):` to not leave the credentials around in long running python sessions.

Reply:
Copying over here from my review. Same concern, different suggestion (although mine would be for later, since it's a change in core)

This is:
1. "Leaking" the credentials into the user's environment.
2. Possibly overwriting existing user settings for use with another webdav special remote.

I think, we would want to allow `AnnexRepo.call_annex` to take an `env` parameter and pass it down instead, patching `env` for the resulting subprocess only.

Reply:
I think this is the wrong place for this comment. Adding support for this might be good, but it should be done in core, and can only later be adopted here. It is also not totally straightforward. The current core runner already supports ENV manipulation, but in doing so already performs several copies of entire environment. "Patching" the environment is something we need to do rather often, but passing entire environment (what we support) is not very useful, in general. What you are asking for will likely result in user code "patching" the environment, in order to pass a full environment to `call_annex`, which passes it to the runner. If that is the case, there is no gain.

Reply:
done